Sources/_StringProcessing/Compiler.swift

```diff
@@ -42,11 +46,6 @@ class Compiler {
   func emit(_ node: AST) throws {
 
     switch node {
-    // Any: .
-    //     consume 1
-    case .atom(let a) where a.kind == .any && matchLevel == .graphemeCluster:
-      builder.buildAdvance(1)
-
```

Comment:
The behavior of "any" depends of the current matching options, so we can't just advance here.

Reply:
In addition to the semantic level, this also needs to take into account whether the regex is in single-line mode — if not, `.` should match line breaks, so there's two levels of checks to make here. In single-line mode we could use the `advance` approach, though.

Reply:
Let's go with the call-out to a closure for now, and we can add real engine support in the future.